client/src/components/Message.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { inject, observer } from "mobx-react";
+import MainStore, { IMessage } from "../store/MainStore";
+import { computed } from "mobx";
+
+export interface IMessageProps {
+    message: IMessage
+}
+
+@inject("store")
+@observer
+export default class Message extends React.Component<IMessageProps> {
+  mainStore = this.injected.store;
+
+  private get injected() {
+    return this.props as IMessageProps & { store: MainStore };
+  }
+
+  @computed
+  private get isOutgoing() {
+      const { clientId } = this.mainStore
+      const { message } = this.props;
+      return message.userId === clientId
+  }
+
+  @computed
+  private get currentDate() {
+    const { message } = this.props;
+    const date = new Date(message.timeStamp)
```

Comment:
Weird destructuring, it's better either
```suggestion
    const { timeStamp} = this.props.message;
    const date = new Date(timeStamp)
```
or
```suggestion
    const date = new Date(this.props.message.timeStamp)
```

Reply:
ok, have done by the first way